compose/ui/ui/src/skikoMain/kotlin/androidx/compose/ui/ComposeScene.skiko.kt

```diff
@@ -607,34 +620,59 @@ class ComposeScene internal constructor(
             previousPressOwner.processPointerInput(event)
             return
         }
+        val position = event.pointers.first().position
         forEachOwnerReversed { owner ->
-            if (owner.isHovered(event)) {
-                // Stop once the position of in bounds of the owner
+
+            // If the position of in bounds of the owner - send event to it and stop processing
+            if (owner.isInBounds(position)) {
                 owner.processPointerInput(event)
                 pressOwner = owner
-                return@processPress
+                return
             }
-            owner.onClickOutside?.invoke()
+
+            // Input event is out of bounds - send click outside notification
+            owner.onOutsidePointerEvent?.invoke(event)
+
+            // if the owner is in focus, do not pass the event to underlying owners
             if (owner == focusedOwner) {
-                // Stop if it's in focus, do not pass the event to hovered owner
-                return@processPress
+                return
             }
         }
     }
 
     private fun processRelease(event: PointerInputEvent) {
-        // Send Release to pressOwner even if is not hovered or under focused.
+        // Send Release to pressOwner even if is not hovered or under focused
         pressOwner?.processPointerInput(event)
+
+        // Process further only if last pointer was released or had pressOwner
+        if (event.isAnyPointerDown || pressOwner != null) {
+            return
```

Comment:
It is easier to read without a middle return:
```
    private fun processRelease(event: PointerInputEvent) {
        if (event.isAnyPointerDown || pressOwner != null) {
            pressOwner?.processPointerInput(event) 
        } else if (!isHoveredOwnerInteractive(event)) {
            // If hovered owner is not interactive, then it means that
            // - It's not focusedOwner
            // - It placed under focusedOwner or not exist at all
            // In all these cases the even happened outside focused owner bounds
            focusedOwner?.onOutsidePointerEvent?.invoke(event)
        }
    }
```

Reply:
The check of `isAnyPointerDown` is incorrect in that snippet. I'll think how to simplify it

Reply:
Extracted condition logic into local function to keep readable comments to all parts of it